### ISSUE TITLE
Docker issue - Why is net=host required? This prevents ports from being exposed to remote machine. Removing it causes call to getNeighbors to return {"error":"COMMAND getNeighbors is not available on this node","duration":16}

### ISSUE THREAD

**Issue by mr-e-** [iotaledger/iri#410]

**Comment by bluedigits** [comment]
Maybe I can answer this. That's related to how Docker works. If you'd remove --net=host Docker would create it's own network, where the ports within the docker subnetwork and the ports which are published to the host's network _can_ differ. That would cause issues with IRI due to some redirection logic. Additionally, there's a Docker issue which occurs on some systems when exposing and publishing UDP ports, even if `java.net.preferIPv4Stack` is set to `true` the port get's exposed as udp6 which causes trouble sometimes.

**Comment by ixidion** [comment]
If you are looking for a Docker setup without network_mode "host", then checkout my docker setup for iri, ipm, nelson: https://github.com/ixidion/iota-docker

**Comment by sajadghawami** [comment]
Having the same issue... any solutions?

**Comment by ixidion** [comment]
I recommend this image. It doesn’t require net=host
https://github.com/chrootLogin/iota-iri

**Comment by sajadghawami** [comment]
@ixidion thanks!

couldn't find anything in the docs about not having to use net=host...

**Comment by ahab94** [comment]
@mr-e-  can you share your `docker-compose`? Maybe i can help and fix the problem.

**Comment by ahab94** [comment]
@bluedigits, `--net=host` approach is not a very good way to resolve this. It's opting for removing and throwing the lid away when it should be exposed for necessary purposes only. IMO, it should be confined.

**Comment by bluedigits** [comment]
@ahab94 I totally agree with you. But it's up to the IRI developers to fix this. As long as the published ports differ from the exposed ports you will run into those problems. If you want both, calling the API from outside whilst having a dedicated docker network for your container you're going to have a problem. You can try it on your own to see what I mean.

**Comment by ahab94** [comment]
@bluedigits, agree. I have something in mind will try to test it locally. Thanks for the input.

**Comment by ahab94** [comment]
```YAML
FROM maven:3.5-jdk-8 as builder
WORKDIR /iri
COPY . /iri
RUN mvn clean package

FROM openjdk:jre-slim
WORKDIR /iri
COPY --from=builder /iri/target/iri-1.4.1.6.jar iri.jar
COPY logback.xml /iri
VOLUME /iri

EXPOSE 14265
EXPOSE 14777/udp
EXPOSE 15777

CMD ["/usr/bin/java", "-XX:+DisableAttachMechanism", "-Xmx8g", "-Xms256m", "-Dlogback.configurationFile=/iri/conf/logback.xml", "-Djava.net.preferIPv4Stack=true", "-jar", "iri.jar", "-p", "14265", "-u", "14777", "-t", "15777", "--remote", "--remote-limit-api", "\"addNeighbors, removeNeighbors\"", "$@"]
```
^^^ This DockerFile works.

So the problem here is that, in the official Dockerfile 
the ``` "--remote-limit-api", "\"addNeighbors, removeNeighbors, getNeighbors\""``` `getNeighbors` api is limited using the `--remote-limit-api` which limits the api to be only accessed via `loopback` interface. Which is crucial to the privacy of the `node` and the neighbors it's using.

**Comment by ahab94** [comment]
Or you can use `https://github.com/ahab94/Iota-Eggs` container, you can simply use ` $ sudo docker run -e REMOTE_LIMIT_API="addNeighbors, removeNeighbors" -e MAX_PEERS=9 -it iota-eggs:latest`. Visit the repository for more clearity. I am the author.
